### DISCUSSION TITLE
Face recognition doesn't recognize some faces - and hidden faces cannot be unhidden

### DISCUSSION THREAD

**Discussion by mtettke123** [immich-app/immich#3760] [Q&A]
Hi,

I've a somewhat strange problem:
While face recognition worked perfect in the past, I have some photos in the last time, where only a.e. one out of three faces is recognized - although all a clearly visible. I wonder, if
- face recognition failed and detected only one
- the "not recognized" faces are somehow matched to hidden faces and are not shown.

So I ask myself
- is there a way to redo face recogintion only on selected photos?
- is there a way to show, if hidden faces were detected in a photo (maybe an option "show hidden faces, too")?

Glad for any help!!!

**Comment by mtettke123** [comment]
... has no one the same problems? No idea how to solve?

**Reply by akropp** [reply]
Yes, I have the same problem.  Images where two people are clearly visible show only one recognized face.  I suspect what you say it true (the other face is being matched to a hidden person).  Would be great to be able to show the hidden faces.

**Comment by mtettke123** [comment]
I just noticed, there seems to be a bigger problem here. All as "hidden" marked faced do no longer show up (even if I click "show/hide"), so I can't even try to test if this happens because of a match to a hidden face.

I've invested too many hours to name and cluster faces - really don't like to loose all my work when starting face recognition from zero!

Any ideas?

**Comment by drifter75** [comment]
is it a bug?

**Reply by CerealeLion** [reply]
I have noticed the same behavior in some photos, clear faces are present, but not all detected (even if the face is already assigned to a person)

**Reply by dominikheinz** [reply]
I was experiencing a similar problem (see https://github.com/immich-app/immich/discussions/7011).
Based on my experimentation, it seems that the problem was a combination of the minimum person count being on 3, and a face being detected as a different person. For some reason the person was detected as a different one, despite being the same. And the min recognized faces settings being set to 3 caused it to be ignored all together. After changing the min recognized setting to 1, the face was showing up. Maybe try that and see if it helps.

**Reply by aviv926** [reply]
Without information about the version and model you are using it will be difficult to give an exact answer but I will give you a background so you can understand how facial recognition works.

Starting with [version 1.94.0](https://github.com/immich-app/immich/releases#:~:text=please%20watch%20this%20video%20for%20a%20step%2Dby%2Dstep%20visualization) the face clustering is done with DBSCAN (watch [this video](https://www.youtube.com/watch?v=RDZUdRSDOok) for a step-by-step visualization), which means that in order for a face to be a face there must be a minimum that meets the requirement of a face, by default a face core is considered from 3 faces per person and above when a person does not have at least 3 faces he will not appear in "people".

In order to solve this problem you can try to lower the face core value to less than 3 by
Administration -> Settings -> Machine Learning Settings -> Facial Recognition -> MIN RECOGNIZED FACES

After the change you will have to run the job FACIAL RECOGNITION
in "missing" mode.



Regarding your other questions:

> is there a way to redo face recognition only on selected photos?

Not at the moment.

> is there a way to show, if hidden faces were detected in a photo (maybe an option "show hidden faces, too")?

Seems [not](https://discord.com/channels/979116623879368755/991643870523830382/1206881762664255559).
